pulsar-broker/src/main/java/org/apache/pulsar/broker/s3offload/DataBlockHeader.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.s3offload;
+
+import java.io.IOException;
+import java.io.InputStream;
+import org.apache.bookkeeper.common.annotation.InterfaceStability.Unstable;
+import org.apache.pulsar.broker.s3offload.impl.DataBlockHeaderImpl;
+
+/**
+ * The data block header in code storage for each data block
+ *
+ * Currently, It is in format:
+ * [ magic_word -- int ][ block_len -- int ][ first_entry_id  -- long][padding]
+ *
+ * with the size: 4 + 4 + 8 + padding = 128 Bytes
+ */
+@Unstable
+public interface DataBlockHeader {
+
+    /**
+     * Get the length of the block in bytes, including the header.
```

Comment:
Should this size include the header? If I have an inputstream, and im reading in, it would seem more natural, that after I've read in the header, the block length that I have is just the block length of the payload.

Reply:
Thanks. This is following the design doc, It would be helpful updating the doc if the idea changed. 
And it seems more natural  to me to make blockLength means the whole block length, since header and payload stored in the same block.

Reply:
Ok. Lets leave it for now, and change it if we need to when we use it.